receiver/statsdreceiver/README.md

```diff
@@ -12,27 +12,46 @@ The following settings are required:
 
 - `endpoint` (default = `localhost:8125`): Address and port to listen on.
 
+
+The Following settings are optional:
+
+- `aggregation_interval: 70s`(default value is 60s): The aggregation time that the receiver aggregates the metrics
+
 Example:
 
 ```yaml
 receivers:
   statsd:
   statsd/2:
     endpoint: "localhost:8127"
+    aggregation_interval: 70s
 ```
 
 The full list of settings exposed for this receiver are documented [here](./config.go)
 with detailed sample configurations [here](./testdata/config.yaml).
 
 ## Aggregation
 
-Currently the `statsdreceiver` is not providing any aggregation. There are
-ideas such as the [Metrics Transform Processor
-Proposal](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/332)
-that intend to enable control over Metric aggregation in a processor.
-
-An alternative will be to implement some simple aggregation in this receiver.
-
+In this new revision. Aggregation is done in statsD receiver. The default aggregation interval is 60s. The receiver only aggregates the metrics with the same metric name, metric type, label keys and label values. After each aggregation interval, the receiver will send all metrics(after aggregation) in this aggregation interval to the following workflow.
+
+It supports:
+
+Gauge:
+- statsdTestMetric1:500|g|#mykey:myvalue
+statsdTestMetric1:400|g|#mykey:myvalue
+(get the latest value: 400)
+- statsdTestMetric1:500|g|#mykey:myvalue
+statsdTestMetric1:+2|g|#mykey:myvalue
+statsdTestMetric1:-1|g|#mykey:myvalue
+(get the value after calculation: 501)
+
+Counter:
+- statsdTestMetric1:3000|c|#mykey:myvalue
+statsdTestMetric1:4000|c|#mykey:myvalue
+(get the value after incrementation: 7000)
+- statsdTestMetric1:3000|c|#mykey:myvalue
+statsdTestMetric1:20|c|@0.8|#mykey:myvalue
+(get the value after incrementation with sample rate: 3000+20/0.8=3025)
```

Comment:
should we mention that it does rounding if needed when there is a sample rate (it does that, right?)

Reply:
Yes, it does rounding. Will revise

Reply:
There's no rounding in this example. Is rounding necessary? I can imagine scenarios where rounding results in various sorts of bias over longer time intervals.

Reply:
If we have "test.metric:42|c|@0.8|#key:value", it will pass the value 52(42/0.8=52.5). Is it okay?

Reply:
I might be forgetting things, but I thought statsd counters have to be integers? I need to check the notes, I thought in our discussions we agreed that weird sampling rates should result in rounding.

Reply:
Hmm I can not find any notes on that, I may or may not have hallucinated that discussion.

Reply:
Yes, we had the agreement that use int for counters and use double for gauges.

Reply:
I feel like this deserves a bit of justification. Personally, I'd choose the `double` representation because of the Statsd sample-rate parameter. If you have a count of 1 and a sample-rate of 0.3 then rounding 3.33333333333 to 3 will under-count by precisely 0.33333333333.

Reply:
I see. Now I used double for gauge and int for counter(after conversion). You suggestion is using double for all all them?

Reply:
We resolved to use double for counters in general, to allow for sampling rate calculations to not lose precision. This is aligned with Prometheus which turns all counters into floating points too.

Reply:
Actually I'm willing to back down here. Sorry!

Reply:
Let's keep integer value counters. When sample-rate is present, I suggest we perform int64(float64(value)/sampleRate), i.e. round the value.

Add a comment somewhere stating that when sample rate is present and not equal to 1.0, the effect will be to convert the count value to float, divide by the sample rate, and convert back to and integer. 

This implies that there are good choices and bad choices of sample rate. Ideally, choose a sample rate that can be exactly expressed in one or two floating point digits where its inverse is exactly an integer. Good sample-rate values: 0.1, 0.2, 0.25, 0.5. Bad sample-rate values: 0.4, 0.3, etc.

Reply:
Good suggestion! Thanks